service/src/main/java/com/codeforcommunity/processor/BlocksProcessorImpl.java

```diff
@@ -109,6 +114,36 @@ public BlockResponse resetBlocks(JWTData jwtData, List<String> blockIds) {
         .collect(Collectors.toList());
   }
 
+  @Override
+  public GetAssignedBlocksResponse getAllReservedBlocks(JWTData jwtData) {
+    return getAssignedBlocksWithStatus(jwtData, BlockStatus.RESERVED);
+  }
+
+  @Override
+  public GetAssignedBlocksResponse getAllDoneBlocks(JWTData jwtData) {
+    return getAssignedBlocksWithStatus(jwtData, BlockStatus.DONE);
+  }
+
+  private GetAssignedBlocksResponse getAssignedBlocksWithStatus(
+      JWTData jwtData, BlockStatus status) {
+    if (jwtData.getPrivilegeLevel() != PrivilegeLevel.ADMIN) {
+      throw new AdminOnlyRouteException();
+    }
+    List<Record3<String, String, Timestamp>> blockRecords =
+        db.select(BLOCK.FID, USERS.USERNAME, BLOCK.UPDATED_TIMESTAMP)
+            .from(BLOCK)
+            .innerJoin(USERS)
+            .on(BLOCK.ASSIGNED_TO.eq(USERS.ID))
+            .where(BLOCK.STATUS.equal(status))
+            .orderBy(BLOCK.UPDATED_TIMESTAMP.desc())
+            .fetch();
```

Comment:
You can call `fetchInto(AssignedBlock.class)` here instead of having to map after fetching